wpilibc/src/main/native/include/frc/shuffleboard/ShuffleboardWidget.inc

```diff
@@ -0,0 +1,24 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+namespace frc {
+
+template <typename Derived>
+ShuffleboardWidget<Derived>::ShuffleboardWidget(ShuffleboardContainer& parent,
+                                                wpi::StringRef title)
+    : ShuffleboardValue(title),
+      ShuffleboardComponent<ShuffleboardWidget<Derived>>(parent, title) {}
+
+template <typename Derived>
+Derived& ShuffleboardWidget<Derived>::WithWidget(wpi::StringRef widgetType) {
+  this->SetType(widgetType);
+  return *static_cast<Derived*>(this);
+}
```

Comment:
These seem like they could be inline, especially since they have to be in headers anyway due to being a template.

Reply:
The point of doing this is to separate interface from implementation the same way .h and .cpp does. I guess they're small enough that merging them into the .h won't affect readability?

Reply:
I generally find for one or two line functions it's fairly clean to have them inline to the class.  Note the way they're defined here they're not actually inline (you would have to specify the "inline" keyword).